packages/core/src/dialects/abstract/data-types.ts

```diff
@@ -2365,6 +2365,12 @@ export class ARRAY<T extends AbstractDataType<any>> extends AbstractDataType<Arr
   ): obj is ARRAY<T> {
     return obj instanceof ARRAY && (obj).options.type instanceof type;
   }
+
+  clone(): this {
+    // there is a convention that all DataTypes must accept a single "options" parameter as one of their signatures, but it's impossible to enforce in typing
+    // @ts-expect-error -- see ^
+    return this._construct(this.options.type.clone().toDialectDataType(this._getDialect()));
```

Comment:
clone should not replace the subtype implementation with the one specific to the dialect. Clone can be called at any point, including before the dialect is available.

toDialectDataType will be called on the subtype when it's called on the array type

Reply:
Unfortunately I don't understand the intricacies of this code change or why it's wrong to use here. It fixed the tests at least.
What would you suggest instead?

Reply:
@ephys still waiting for help on this one.